library/src/scala/quoted/Expr.scala

```diff
@@ -184,8 +184,44 @@ package quoted {
       val elems: Seq[Expr[_]] = tup.asInstanceOf[Product].productIterator.toSeq.asInstanceOf[Seq[Expr[_]]]
       ofTuple(elems).cast[Tuple.InverseMap[T, Expr]]
     }
-  }
 
+    // TODO generalize for any function arity (see Expr.betaReduce)
+    def open[T1, R, X](f: Expr[T1 => R])(content: (Expr[R], [t] => Expr[t] => Expr[T1] => Expr[t]) => X)(given qctx: QuoteContext): X = {
+      import qctx.tasty.{given, _}
+      val (params, bodyExpr) = paramsAndBody(f)
+      content(bodyExpr, [t] => (e: Expr[t]) => (v: Expr[T1]) => bodyFn[t](e.unseal, params, List(v.unseal)).seal.asInstanceOf[Expr[t]])
+    }
```

Comment:
It's not clear how `open/close` is useful. I'd suggest removing them for now.

Reply:
I cannot remove it without breaking some tests. I particular those that test the code infrastructure for what @biboudis is working on.